PlaywrightDemoQA/WebTablesTest.cs

```diff
@@ -0,0 +1,86 @@
+﻿using Microsoft.Playwright;
+
+namespace PlaywrightTests;
+
+[Parallelizable(ParallelScope.Self)]
+[TestFixture]
+public class WebTablesTest
+{
+    [Test]
+    public async Task DeleteRow()
+    {
+        using var playwright = await Playwright.CreateAsync();
+        await using var browser = await playwright.Chromium.LaunchAsync(new BrowserTypeLaunchOptions
+        {
+            Headless = false,
+        });
+        var context = await browser.NewContextAsync();
+
+        var page = await context.NewPageAsync();
+        await page.GotoAsync("https://demoqa.com/webtables");
+        await page.Locator(".rt-tr-group")
+            .Filter(new() { Has = page.GetByRole(AriaRole.Gridcell, new() { Name = "Alden", Exact = true }) })
+            .GetByTitle("Delete").ClickAsync();
+        await Assertions.Expect(page.GetByText("Alden")).Not.ToBeVisibleAsync();
+    }
+
+    [Test]
+    public async Task EditLastNameandSallary()
+    {
+        using var playwright = await Playwright.CreateAsync();
+        await using var browser = await playwright.Chromium.LaunchAsync(new BrowserTypeLaunchOptions
+        {
+            Headless = false,
+        });
+        var context = await browser.NewContextAsync();
+
+        var page = await context.NewPageAsync();
+        await page.GotoAsync("https://demoqa.com/webtables");
+        await page.Locator(".row > div:nth-child(3)").ClickAsync();
+          await page.Locator("#edit-record-2 path").ClickAsync();
+        await page.GetByPlaceholder("Last Name").ClickAsync();
+        await page.GetByPlaceholder("Last Name").FillAsync("Cantrell edit");
+        await page.GetByPlaceholder("Salary").FillAsync("15000");
+        await page.GetByRole(AriaRole.Button, new() { Name = "Submit" }).ClickAsync();
+        await Assertions.Expect(page.Locator(".col-12 > div:nth-child(4)")).ToBeVisibleAsync();
+
+
+        await Assertions.Expect(page.GetByRole(AriaRole.Grid)).ToContainTextAsync("Cantrell edit");
+
+        await Assertions.Expect(page.GetByRole(AriaRole.Grid)).ToContainTextAsync("15000");
+    }
+
+    [Test]
+    public async Task AddNewRecord()
+    {
+        using var playwright = await Playwright.CreateAsync();
+        await using var browser = await playwright.Chromium.LaunchAsync(new BrowserTypeLaunchOptions
+        {
+            Headless = false,
+        });
+        var context = await browser.NewContextAsync();
+
+        var page = await context.NewPageAsync();
+        await page.GotoAsync("https://demoqa.com/webtables");
+        await page.Locator(".row > div:nth-child(3)").ClickAsync();
```

Comment:
what this line does? Test should add new row. Button "Add" is clicked on the next line. What this click is needed for?